File: OpenRobertaServer/src/test/resources/crossCompilerTests/_expected/robotSpecific/targetLanguage/calliope2016/callibot_motor.cpp
#define _GNU_SOURCE

#include "MicroBit.h"
#include "NEPODefs.h"
#include <list>
#include <array>
#include <stdlib.h>
MicroBit _uBit;
MicroBitI2C _i2c(MICROBIT_PIN_P20, MICROBIT_PIN_P19);
char _buf[5] = { 0, 0, 0, 0, 0 };
uint8_t _cbLedState = 0x00;
void ____wait();



int main()
{
    _uBit.init();
    _cbInit(_buf, &_i2c, &_uBit);

    _uBit.display.scroll(ManagedString("Curve Right"));
    _cbSetMotors(_buf, &_i2c, 30, 10);
    ____wait();
    _uBit.display.scroll(ManagedString("Curve Left"));
    _cbSetMotors(_buf, &_i2c, 10, 30);
    ____wait();
    _uBit.display.scroll(ManagedString("Forward"));
    _cbSetMotors(_buf, &_i2c, 100, 100);
    ____wait();
    _uBit.display.scroll(ManagedString("Backwards"));
    _cbSetMotors(_buf, &_i2c, -100, -100);
    ____wait();
    _uBit.display.scroll(ManagedString("Turn Left"));
    _cbSetMotors(_buf, &_i2c, -50, 50);
    ____wait();
    _uBit.display.scroll(ManagedString("Turn Right"));
    _cbSetMotors(_buf, &_i2c, 50, -50);
    ____wait();
    _uBit.display.scroll(ManagedString("STOP"));
    _cbSetMotors(_buf, &_i2c, 0, 0);
    ____wait();
    _uBit.display.scroll(ManagedString("Left On"));
    _cbSetMotor(_buf, &_i2c, 0, 30);
    ____wait();
    _uBit.display.scroll(ManagedString("Stop Left"));
    _cbSetMotor(_buf, &_i2c, 0, 0);
    ____wait();
    _uBit.display.scroll(ManagedString("Right On"));
    _cbSetMotor(_buf, &_i2c, 0, 30);
    ____wait();
    _uBit.display.scroll(ManagedString("Stop Right"));
    _cbSetMotor(_buf, &_i2c, 2, 0);
    ____wait();
    _uBit.display.scroll(ManagedString("Servo 90"));
    _cbSetServo(_buf, &_i2c, 0x14, 90);
    ____wait();
    _uBit.display.scroll(ManagedString("Servo 45"));
    _cbSetServo(_buf, &_i2c, 0x15, 45);
    _cbStop(_buf, &_i2c);
    release_fiber();
}

void ____wait() {
    while (true) {
        if ( _uBit.buttonA.isPressed() == true ) {
            break;
        }
        _uBit.sleep(_ITERATION_SLEEP_TIMEOUT);
    }
    _uBit.sleep(500);
}

File: OpenRobertaServer/src/test/resources/crossCompilerTests/_expected/robotSpecific/targetLanguage/calliopev3/callibot_motor.py
import calliopemini
import random
import math
from callibot2 import Callibot2

class BreakOutOfALoop(Exception): pass
class ContinueLoop(Exception): pass

timer1 = calliopemini.running_time()
callibot = Callibot2()


def ____wait():
    global timer1
    while True:
        if calliopemini.button_a.is_pressed() == True:
            break
    calliopemini.sleep(500)


def run():
    global timer1
    calliopemini.display.scroll("Curve Right")
    callibot.set_speed(30, 10)
    ____wait()
    calliopemini.display.scroll("Curve Left")
    callibot.set_speed(10, 30)
    ____wait()
    calliopemini.display.scroll("Forward")
    callibot.set_speed(100, 100)
    ____wait()
    calliopemini.display.scroll("Backwards")
    callibot.set_speed(-100, -100)
    ____wait()
    calliopemini.display.scroll("Turn Left")
    callibot.set_speed(-50, 50)
    ____wait()
    calliopemini.display.scroll("Turn Right")
    callibot.set_speed(50, -50)
    ____wait()
    calliopemini.display.scroll("STOP")
    callibot.stop()
    ____wait()
    calliopemini.display.scroll("Left On")
    callibot.set_speed_motor(1, 30)
    ____wait()
    calliopemini.display.scroll("Stop Left")
    callibot.stop_motor(1)
    ____wait()
    calliopemini.display.scroll("Right On")
    callibot.set_speed_motor(1, 30)
    ____wait()
    calliopemini.display.scroll("Stop Right")
    callibot.stop_motor(2)
    ____wait()
    calliopemini.display.scroll("Servo 90")
    callibot.servo(1, 90)
    ____wait()
    calliopemini.display.scroll("Servo 45")
    callibot.servo(2, 45)

def main():
    try:
        run()
    except Exception as e:
        raise
    finally:
        callibot.stop()


if __name__ == "__main__":
    main()